File: modul_praktikum_python/contoh_code/python/pangkat_latihan.py
"""Latihan fungsi pangkat - isi fpangkat (stub)."""


def fpangkat(dipangkatkan: float, pangkat: int) -> float:
    # TODO: lengkapi (pahami pangkat_for / modul pangkat)
    raise NotImplementedError("Lengkapi fpangkat seperti di modul Pascal.")


def main() -> None:
    dipangkatkan = float(input("Bilangan yang akan dipangkatkan : "))
    pangkat = int(input("Masukkan pangkatnya : "))
    try:
        print(dipangkatkan, "^", pangkat, " : ", fpangkat(dipangkatkan, pangkat), sep="")
    except NotImplementedError as e:
        print(e)
    input("Tekan Enter untuk melanjutkan...")


if __name__ == "__main__":
    main()

File: modul_praktikum_python/contoh_code/pascal/pangkat_latihan.pas
uses crt; 
function fpangkat(dipangkatkan:real;pangkat:integer):real; 
var  
  (* variabelnya anda tentukan sendiri  *) 
begin 
  (* pahami program pangkat.pas dan kemudian lengkapi bagian 
ini*) 
end; 

var 
  (* lengkapi bagian ini *) 
begin 
  write(‘Bilangan yang akan dipangkatkan : ‘); 
  readln(dipangkatkan); 
  write(‘Masukkan pangkatnya : ‘);readln(pangkat);  
  write(dipangkatkan,‘^’,pangkat,’ : ‘, 
        fpangkat(dipangkatkan,pangkat)); 
  readkey; 
end.
